### DISCUSSION TITLE
Audio Tools with AI Thinker v 2.2 A252

### DISCUSSION THREAD

**Discussion by SnazzyBlanket** [pschatzmann/arduino-audio-tools#1427] [Q&A] [answered]
I am at a loss at the moment as to how to set up this board. I bought it thinking it would be easier to use only to find its a bit of a nightmare of weird versions.

using this code below I can get sound coming out BUT there is def something wrong with it (improper pin or clocking would be my guess)as it is distorted and crackly.

`/**
 * @file streams-synthbasic-audiokit.ino
 * @author Phil Schatzmann
 * @brief see https://www.pschatzmann.ch/home/2021/12/17/ai-thinker-audio-kit-building-a-simple-synthesizer-with-the-audiotools-library/)
 * @copyright GPLv3
 * 
 */
#include <Arduino.h>
#include "WiFi.h"

#include "AudioTools.h"

#include "AudioLibs/AudioBoardStream.h"

AudioBoardStream kit(AudioKitEs8388V2);
SineWaveGenerator<int16_t> sine;
GeneratedSoundStream<int16_t> in(sine); 
StreamCopy copier(kit, in); 

void actionKeyOn(bool active, int pin, void* ptr){
  int freq = *((float*)ptr);
  sine.setFrequency(freq);
  in.begin();
}

void actionKeyOff(bool active, int pin, void* ptr){
  in.end();
}

// We want to play some notes on the AudioKit keys 
void setupActions(){
  // assign buttons to notes
  auto act_low = AudioActions::ActiveLow;
  static float note[] = {N_C3, N_D3, N_E3, N_F3, N_G3, N_A3}; // frequencies
  kit.audioActions().add(kit.getKey(1), actionKeyOn, actionKeyOff, act_low, &(note[0])); // C3
  kit.audioActions().add(kit.getKey(2), actionKeyOn, actionKeyOff, act_low, &(note[1])); // D3
  kit.audioActions().add(kit.getKey(3), actionKeyOn, actionKeyOff, act_low, &(note[2])); // E3
  kit.audioActions().add(kit.getKey(4), actionKeyOn, actionKeyOff, act_low, &(note[3])); // F3
  kit.audioActions().add(kit.getKey(5), actionKeyOn, actionKeyOff, act_low, &(note[4])); // G3
  kit.audioActions().add(kit.getKey(6), actionKeyOn, actionKeyOff, act_low, &(note[5])); // A3
}

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial,AudioLogger::Info);

  // Setup output
  auto cfg = kit.defaultConfig(TX_MODE);
  cfg.sd_active = false;
  kit.begin(cfg);
  kit.setVolume(50);

  // Setup sound generation based on AudioKit settins
  in.begin(cfg);

  // activate keys
  setupActions();

}

// copy the data
void loop() {
  copier.copy();
  kit.processActions();
}`

I also tried the other driver options but this was the only one that gave any sound at all

THANKS for any help

**Comment by pschatzmann** [comment, marked as answer]
https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working

Serial output is slow and therefore can have an impact on audio: This is described in multiple places in the documentation!
So just switch the logging to warning (by using AudioLogger::Info) or lower the sample rate(  cfg.sample_rate = 22050;).

**Comment by SnazzyBlanket** [comment]
Ahh I was so focused on it possibly being one of the AI thinker boards with issues I did not even think to switch to warning or error (which I have read about before )thanks Sent from my iPhoneOn Mar 7, 2024, at 10:22 AM, Phil Schatzmann ***@***.***> wrote:﻿
https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working
